### ISSUE TITLE
A11y: TreeView expand/collapse is not accessible for keyboard user

### ISSUE THREAD

**Issue by Tushar2098** [vmware/clarity#6332]
## Describe the bug

TreeView expand/collapse is not accessible for keyboard user

## How to reproduce

1.  Go to [Stackblitz](https://stackblitz.com/edit/clarity-v4-dark-theme-qyszmu)
2.  Turn on the VoiceOver (CMD+F5)
3.  Navigate to root node by pressing TAB
4.  Press ENTER to expand the TreeView
5.  Observe TreeView is not expanding and same goes for collapse

## Expected behavior

User should be able to change expand/collapse state via keyboard. And screen reader should announce the expand/collapse state 

## Versions

**Clarity project:**

- [ ] Clarity Core
- [x] Clarity Angular/UI

**Clarity version:**

- [ ] v3.x
- [x] v4.x
- [ ] v5.x

**Framework:**

- [x] Angular
- [ ] React
- [ ] Vue
- [ ] Other:

**Framework version:**
_ie: Angular 11_

**Device:**

- Type: [e.g. MacBook]

## Additional notes

_Add any other notes about the problem here._

**Comment by bdryanovski** [comment]
@Tushar2098 - thanks for reporting this we gonna investigate it a little bit more.

**Comment by Jinnie** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally.
The Clarity Core tree has better handling of Enter and Space keys, as can be seen from this example:
https://clarity.design/storybook/core/?path=/story/stories-tree-view--interactive
It is currently released in latest Clarity Core releases, but it is still a BETA version.
We'll further investigate if we can deliver this fix for Clarity Angular too.
